src/NetTopologySuite.IO.Esri/Readers/ShapefileReader.cs

```diff
@@ -0,0 +1,296 @@
+﻿using NetTopologySuite.Features;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.IO.Dbf;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace NetTopologySuite.IO.Shapefile
+{
+    /// <summary>
+    /// Base class for reading a shapefile.
+    /// </summary>
+    public abstract class ShapefileReader : Shapefile, IEnumerable<Feature>
```

Comment:
This class needs a `GeometryFactory` instance member that is accessible by the derived classes.   
Instead of instantiating geometries with their constructors (e.g. `new Point(x, y, z, m)` geometries should be instantiated using the appropriate `GeometryFactory.Create...` method.